src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -3097,7 +3130,7 @@ def aks_agentpool_add(cmd, client, resource_group_name, cluster_name, nodepool_n
                       vnet_subnet_id=None,
                       ppg=None,
                       max_pods=0,
-                      os_type="Linux",
+                      os_type=None,
```

Comment:
Same as above

Reply:
the os_type still defaults to "Linux" on Ln 3186 if customer doesn't specify os_type or not specify a snapshot

Reply:
Similar to the above, since we have a new source of this parameter value, we have to set the default parameter to `None` to help us determine whether the user provides `"Linux"` as input (the former default value), or does not provide any input.